akka-actor/src/main/scala/akka/dispatch/BatchingExecutor.scala

```diff
@@ -89,21 +89,22 @@ private[akka] trait BatchingExecutor extends Executor {
         parentBlockContext = prevBlockContext
         try processBatch(this) catch {
           case t: Throwable ⇒
+            parentBlockContext = null
             resubmitUnbatched()
             throw t
-        } finally {
-          _tasksLocal.remove()
-          parentBlockContext = null
-        }
+        } finally _tasksLocal.remove()
+        parentBlockContext = null
```

Comment:
Yes, very much intentional: we _must not_ modify `parentBlockContext` after `resubmitUnbatched()`, I’ll add a comment.